.golangci.yaml

```diff
@@ -1,120 +1,81 @@
-  misspell:
-    locale: US
-  nolintlint:
-    allow-unused: false # report any unused nolint directives
-    require-explanation: false # don't require an explanation for nolint directives
-    require-specific: false # don't require nolint directives to be specific about which linter is being skipped
+version: "2"
 
 linters:
-  # please, do not use `enable-all`: it's deprecated and will be removed soon.
-  # inverted configuration with `enable-all` and `disable` is not scalable during updates of golangci-lint
-  disable-all: true
-  enable:
-    - asciicheck
-    - bodyclose
-    - dogsled
+  default: all
+
+  # explicitly disabled linters
+  disable:
+    - containedctx    
+    - contextcheck
+    - cyclop
+    - depguard
     - errcheck
-    - copyloopvar
+    - exhaustruct
     - exhaustive
-    - funlen
-    - gochecknoinits
-    - goconst
-    - gocritic
-    - gocyclo
-    - godot
-    - err113
-    - gofmt
-    - goimports
-    - goprintffuncname
-    - gosec
-    - gosimple
-    - govet
-    - ineffassign
-    - misspell
-    - nakedret
-    - noctx
-    - rowserrcheck
-    - staticcheck
-    - stylecheck
-    - testpackage
-    - typecheck
-    - unconvert
-    - unparam
-    - unused
-
-  # don't enable:
-    # - depguard
-    # - dupl
-    # - gochecknoglobals
-    # - gocognit
-    # - godox
-    # - gomnd
-    # - lll
-    # - nestif
-    # - nolintlint # conflict with 1.19 gofmt changes
-    # - prealloc
-    # - revive
-    # - wsl
-    # - whitespace
+    - forbidigo
+    - gomodguard # no go.mod at root level    
+    - gochecknoglobals  # no configuration options
+    - nilnil
+    - nlreturn          # redundant with wsl
+    - paralleltest
+    - revive
+    - tagalign
+    - thelper
+    - varnamelen
+    - wrapcheck
 
-issues:
-  # List of regexps of issue texts to exclude, empty list by default.
-  # But independently from this option we use default exclude patterns,
-  # it can be disabled by `exclude-use-default: false`. To list all
-  # excluded by default patterns execute `golangci-lint run --help`
-  exclude:
-  - declaration of "(err|ctx)" shadows declaration at
-  - shadow of imported from 'github.com/stretchr/testify/assert' package 'assert'
-  # Excluding configuration per-path, per-linter, per-text and per-source
-  exclude-rules:
-    - path: _test\.go
-      linters:
-      - gomnd
-    # https://github.com/go-critic/go-critic/issues/926
-    - text: "unnecessaryDefer:"
-      linters:
-      - gocritic
-    - text: "unnamedResult:"
-      linters:
-      - gocritic
-    - text: "G404"
-      linters:
-      - gosec
\ No newline at end of file
+  settings:
+    ireturn:
+      allow:
+        - error
+        - empty
+        - generic
+        - proto.Message
+        - plugins.Plugin
+        - stdlib
+        - v3.ImporterClient
+    gomoddirectives:
+      replace-local: true
+    lll:
+      line-length: 140 # a lot of parameter lines over 130 characters
+    dupl:
+      threshold: 100
+    funlen:
```

Comment:
Why increase  `funlen` limits even more? We already extend the default limits to 80/60 in other projects, why do they need to be even higher here?